client/src/rpc_cache.rs

```diff
@@ -0,0 +1,169 @@
+use crate::rpc_config::RpcLargestAccountsConfig;
+use crate::rpc_response::RpcAccountBalance;
+use std::collections::HashMap;
+use std::time::{Duration, SystemTime};
+
+#[derive(Debug, Clone)]
+pub struct LargestAccountsCache {
+    size: usize,
+    duration: u64,
+    cache: HashMap<RpcLargestAccountsConfig, LargestAccountsCacheValue>,
+}
+
+#[derive(Debug, Clone)]
+struct LargestAccountsCacheValue {
+    accounts: Vec<RpcAccountBalance>,
+    cached_time: SystemTime,
+}
+
+impl LargestAccountsCache {
+    pub fn new(size: usize, duration: u64) -> Self {
+        Self {
+            size,
+            duration,
+            cache: HashMap::new(),
+        }
+    }
+
+    pub fn get_largest_accounts(
+        &self,
+        config: &RpcLargestAccountsConfig,
+    ) -> Option<Vec<RpcAccountBalance>> {
+        self.cache.get(&config).map(|value| {
+            value.cached_time.elapsed().ok().map(|elapsed| {
+                if elapsed < Duration::from_secs(self.duration) {
+                    Some(value.accounts.clone())
+                } else {
+                    None
+                }
+            })?
+        })?
+    }
+
+    pub fn set_largest_accounts(
+        &mut self,
+        config: &RpcLargestAccountsConfig,
+        accounts: &[RpcAccountBalance],
+    ) {
+        if self.cache.len() >= self.size {
+            self.evict();
```

Comment:
Not sure this (or the size field, for that matter) is necessary if we key by `Option<RpcLargestAccountsFilter>`, is it? There will only ever be max 3 HashMap records.

Reply:
Yep, we can do away with this now! Mostly a holdover from when I wrote this for program accounts.